src/classes/window/fullscreen.vala

```diff
@@ -79,6 +84,8 @@ namespace pdfpc.Window {
                 screen.get_monitor_geometry(current_screen, out this.screen_geometry);
             }
 
+            this.gdk_scale = screen.get_monitor_scale_factor((screen_num == -1) ? 0 : screen_num);
```

Comment:
I think that this should default to `current_screen` (currently out of scope, defined in the `if` directly above this line) instead of screen `0`.